### PR TITLE
Open parallel cursors on behalf of the current user

### PR DESCRIPTION
**Pull request by adam8157**
It's a weird behavior that parallel retrieve cursors are opened on
behalf of the session user no matter who the current user is, and it
confuses the database users.

This commit changes the behavior to open parallel retrieve cursors on
behalf of the current user, just as how these parallel retrieve cursor
functions check permissions.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [x] Pass `make installcheck`

### CODE REVIEW

File: src/backend/cdb/endpoint/cdbendpoint.c [hunk 1]
```diff
@@ -486,7 +486,7 @@ static Endpoint
 	StrNCpy(sharedEndpoints[i].cursorName, cursorName, NAMEDATALEN);
 	sharedEndpoints[i].databaseID = MyDatabaseId;
 	sharedEndpoints[i].sessionID = gp_session_id;
-	sharedEndpoints[i].userID = GetSessionUserId();
```
Comment (zxuejing) [inline src/backend/cdb/endpoint/cdbendpoint.c:489]: whether give a notice or throw an error when user who without login privilege create a parallel retrieve cursor?

Comment (adam8157) [reply src/backend/cdb/endpoint/cdbendpoint.c:489]: It's probably not needed I suppose.

Before this PR, a user without login privilege could be set as the session user via `SET SESSION AUTHORIZATION` and create parallel retrieve cursors, we didn't notice or error out either. They will find out when they try to establish retrieve connections.

**Review by zxuejing — APPROVED** [review]

**Comment by adam8157** [comment]
@dyozie This PR changed the behavior documented in `admin_guide-parallel_retrieve_cursor.html`, now it opens parallel cursors on behalf of the current user but not the session user, could you please help review the behavior change and file a doc story if you agree with the change?

The old behavior was weird that in the case below the `SELECT * FROM gp_endpoints;` returns empty.
```sql
SET ROLE user2;
BEGIN; DECLARE prc1 PARALLEL RETRIEVE CURSOR FOR SELECT version();
SELECT * FROM gp_endpoints;
```

**Review by zxuejing — APPROVED** [review]

**Comment by dyozie** [comment]
Thanks @adam8157 - I've created a Jira to address this change in the docs. Do you think this change will stay in main or possibly cherry-pick into 6X_STABLE?

**Comment by adam8157** [comment]
@dyozie yes, we have the PR for 6X https://github.com/greenplum-db/gpdb/pull/15854.

**Pull request merged**